packages/belos/tpetra/example/TFQMR/PseudoBlockTFQMRTpetraExFile.cpp

```diff
@@ -0,0 +1,246 @@
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+//
+// This driver reads a problem from a Harwell-Boeing (HB) file.
+// Multiple right-hand-sides are created randomly.
+// The initial guesses are all set to zero.
+//
+
+// Belos
+#include "BelosConfigDefs.hpp"
+#include "BelosLinearProblem.hpp"
+#include "BelosPseudoBlockTFQMRSolMgr.hpp"
+#include "BelosTpetraAdapter.hpp"
+#include "BelosTpetraTestFramework.hpp"
+
+// Ifpack2
+#include "Ifpack2_Factory.hpp"
+#include "Ifpack2_Preconditioner.hpp"
+
+// Tpetra
+#include "Tpetra_Core.hpp"
+#include "Tpetra_CrsMatrix.hpp"
+#include "Tpetra_Map.hpp"
+#include "Tpetra_MultiVector.hpp"
+#include "Tpetra_Operator.hpp"
+
+// Teuchos
+#include "Teuchos_Assert.hpp"
+#include "Teuchos_CommandLineProcessor.hpp"
+#include "Teuchos_ParameterList.hpp"
+#include "Teuchos_StandardCatchMacros.hpp"
+
+template <typename ScalarType>
+int run(int argc, char *argv[]) {
+  using Teuchos::ParameterList;
+  using Teuchos::RCP;
+  using Teuchos::rcp;
+
+  using ST = typename Tpetra::MultiVector<ScalarType>::scalar_type;
+  using LO = typename Tpetra::Vector<>::local_ordinal_type;
+  using GO = typename Tpetra::Vector<>::global_ordinal_type;
+  using NT = typename Tpetra::Vector<>::node_type;
+
+  using MV = typename Tpetra::MultiVector<ST, LO, GO, NT>;
+  using OP = typename Tpetra::Operator<ST, LO, GO, NT>;
+  using MAP = typename Tpetra::Map<LO, GO, NT>;
+  using MAT = typename Tpetra::CrsMatrix<ST, LO, GO, NT>;
+
+  using MVT = typename Belos::MultiVecTraits<ST, MV>;
+  using OPT = typename Belos::OperatorTraits<ST, MV, OP>;
+
+  using MT = typename Teuchos::ScalarTraits<ST>::magnitudeType;
+
+  using LinearProblem = typename Belos::LinearProblem<ST, MV, OP>;
+  using Preconditioner = typename Ifpack2::Preconditioner<ST, LO, GO, NT>;
+  using Solver = ::Belos::PseudoBlockTFQMRSolMgr<ST, MV, OP>;
+
+  Teuchos::GlobalMPISession session(&argc, &argv, NULL);
+  RCP<const Teuchos::Comm<int>> comm = Tpetra::getDefaultComm();
+
```

Comment:
You can define here: `int MyPID = rank(*comm);` and use this `MyPID` line 130

Reply:
I think it is not necessary as it is required just once in the code